manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/MatchTest.scala

```diff
@@ -19,347 +19,305 @@
-      optionalResultExplanation = """Returns nodes that +ACTED_IN+ Wall Street.""",
-      assertions = (p) => assertEquals(3, p.size)
-    )
-  }
-
-  @Test def relationshipsByTypeWithSpace() {
-    testQuery(
-      title = "Relationship types with uncommon characters",
-      text = "Sometime your database will have types with non-letter characters, or with spaces in them. Use +`+ (backtick) to quote these.",
-      queryText = """match (n {name:'Rob Reiner'})-[r:`TYPE THAT HAS SPACE IN IT`]->() return r""",
-      optionalResultExplanation = """Returns a relationship of a type with spaces in it.""",
-      assertions = (p) => assertEquals(1, p.size)
-    )
-  }
-
-  @Test def multiStepRelationships() {
-    testQuery(
-      title = "Multiple relationships",
-      text = "Relationships can be expressed by using multiple statements in the form of `()--()`, or they can be strung together, " +
-        "like this:",
-      queryText = """match (charlie {name:'Charlie Sheen'})-[:ACTED_IN]->(movie)<-[:DIRECTED]-(director) return charlie,movie,director""",
-      optionalResultExplanation = """Returns the three nodes in the path.""",
-      assertions = (p) => assertEquals(List(Map("charlie" -> node("Charlie"), "movie" -> node("WallStreet"), "director" -> node("Oliver"))), p.toList)
-    )
-  }
-
-  @Test def variableLengthPath() {
-    testQuery(
-      title = "Variable length relationships",
-      text = "Nodes that are a variable number of relationship->node hops away can be found using the following syntax: `-[:TYPE*minHops..maxHops]->`. " +
-        "minHops and maxHops are optional and default to 1 and infinity respectively. When no bounds are given the dots may be omitted.",
-      queryText = """match (martin {name:"Martin Sheen"})-[:ACTED_IN*1..2]-(x) return x""",
-      optionalResultExplanation = "Returns nodes that are 1 or 2 relationships away from Martin.",
-      assertions = (p) => assertEquals(Set(node("Charlie"), node("WallStreet"), node("Michael"), node("TheAmericanPresident")), p.columnAs[Node]("x").toSet)
-    )
-  }
-
-  @Test def variableLengthPathWithIterableRels() {
-    testQuery(
-      title = "Relationship identifier in variable length relationships",
-      text = "When the connection between two nodes is of variable length, " +
-        "a relationship identifier becomes an collection of relationships.",
-      queryText = """match (actor {name:'Charlie Sheen'})-[r:ACTED_IN*2]-(co_actor) return r""",
-      optionalResultExplanation = "The query returns a collection of relationships.",
-      assertions = (p) => assertEquals(2, p.size)
-    )
-  }
+      p("The following graph is used for the examples below:")
+      graphViz()
+    }
+    section("Basic node finding") {
+      section("Get all nodes") {
+        p("By just specifying a pattern with a single node and no labels, all nodes in the graph will be returned.")
+        query("MATCH (n) RETURN n", assertAllNodesReturned) {
```

Comment:
This isn't easy. As this is the first example on the page, it should ideally be both a super simple query _and_ very clean output.

As the next example does return a property and not full nodes I think we can keep this one as it is.